Formula/php.rb

```diff
@@ -0,0 +1,333 @@
+      --with-imap-ssl=#{Formula["openssl"].opt_prefix}
+      --with-libzip=#{Formula["libzip"].opt_prefix}
+      --with-mhash
+      --with-mysql-sock=/tmp/mysql.sock
+      --with-mysqli=mysqlnd
+      --with-pdo-mysql=mysqlnd
+      --with-pdo-odbc=unixODBC,#{Formula["unixodbc"].opt_prefix}
+      --with-ndbm=/usr
+      --with-openssl=#{Formula["openssl"].opt_prefix}
+      --with-pdo-dblib=#{Formula["freetds"].opt_prefix}
+      --with-pdo-pgsql=#{Formula["libpq"].opt_prefix}
+      --with-pgsql=#{Formula["libpq"].opt_prefix}
+      --with-png-dir=#{Formula["libpng"].opt_prefix}
+      --with-pspell=#{Formula["aspell"].opt_prefix}
+      --with-snmp
+      --with-tidy=shared,#{Formula["tidy-html5"].opt_prefix}
+      --with-unixODBC=#{Formula["unixodbc"].opt_prefix}
+      --with-webp-dir=#{Formula["webp"].opt_prefix}
+      --with-pic
+      --with-xmlrpc
+      --with-zlib=/usr
+      --with-libedit
+      --with-xsl=/usr
+    ]
+
+    if MacOS.version < :lion
+      args << "--with-curl=#{Formula["curl"].opt_prefix}"
+    else
+      args << "--with-curl"
+    end
+
+    args << "--enable-maintainer-zts" if build.with? "thread-safety"
+
+    if build.devel?
+      args += %W[
+        --with-password-argon2=#{Formula["argon2"].opt_prefix}
+        --with-sodium=#{Formula["libsodium"].opt_prefix}
+      ]
+    else
+      args << "--with-mcrypt=shared,#{Formula["mcrypt"].opt_prefix}"
+    end
+
+    system "./configure", *args
+
+    inreplace "Makefile" do |s|
+      # Custom location for php module and remove -a (don't touch httpd.conf)
+      s.gsub! /^INSTALL_IT = \$\(mkinstalldirs\) '([^']+)' (.+) LIBEXECDIR=([^\s]+) (.+) -a (.+)$/,
+        "INSTALL_IT = $(mkinstalldirs) '#{lib}/httpd/modules' \\2 LIBEXECDIR='#{lib}/httpd/modules' \\4 \\5"
+
+      # Reorder linker flags to put system paths at the end to avoid accidential system linkage
+      %w[EXTRA_LDFLAGS EXTRA_LDFLAGS_PROGRAM].each do |mk_var|
+        system_libs = []
+        other_flags = []
+        s.get_make_var(mk_var).split.each do |f|
+          if f[%r{^-L\/(?:Applications|usr\/lib)\/}]
+            system_libs << f
+            next
+          end
+          other_flags << f
+        end
+        s.change_make_var! mk_var, other_flags.concat(system_libs).join(" ")
+      end
+    end
+
+    # Shared module linker flags come after the sytem flags, prepend to avoid accidential system linkage
+    ENV.prepend "LDFLAGS", "-L#{Formula["tidy-html5"].opt_lib}"
+
+    system "make"
+    ENV.deparallelize
+    system "make", "install"
+
+    bin.install_symlink "phar.phar" => "phar"
+
+    config_path.install "php.ini-development" => "php.ini"
+    (config_path/"php-fpm.d").install "sapi/fpm/www.conf"
+    config_path.install "sapi/fpm/php-fpm.conf"
+    inreplace config_path/"php-fpm.conf", /^;?daemonize\s*=.+$/, "daemonize = no"
+
+    # patch PEAR so it installs extensions outside of the Cellar
+    Dir.chdir prefix do
+      pear_patch = Patch.create :p1, <<-EOS.undent
```

Comment:
This needs to not be in the formula.

Reply:
We need some better ideas then, because the extension installing binary `pecl` will install shared objects into the Cellar and they'll get lost at version bump.

Reply:
I think we need a configure option from upstream.

Reply:
It's been open upstream for a while, http://pear.php.net/bugs/bug.php?id=18666, I doubt it's going to get an more traction (too much legacy). 

I'll see if there is some other approach I can take by doing some gsub'ing on the build files as was done in httpd.

Reply:
>too much legacy

Can you elaborate? Why are we relying on something upstream considers "too much legacy"? What is the non-legacy approach?

Reply:
Perhaps that was the wrong choice of words. The repurposing of an established, but unused configuration variable `ext_dir` was cause for worry (legacy/regression) and there didn't seem to be any drive to create a new configuration or API.

Reply:
My new approach for this is https://github.com/kabel/homebrew-core/blob/a910b289ba94cf837e4a6c3ad5c89bf51c17ef5e/Formula/php.rb#L240-L255 and https://github.com/kabel/homebrew-core/blob/a910b289ba94cf837e4a6c3ad5c89bf51c17ef5e/Formula/php.rb#L212-L238